swift/multiplay.swift

```diff
@@ -0,0 +1,331 @@
+            if (!session) {
+                session = new MPLocalSession(this.connection, data.localSessionToken, data.displayName, data.shareCode);
+                this.sessions.push(session);
+            }
+            return session;
+        });
+    }
+
+    leaveSession(session) {
+        let sessions = this.sessions;
+        const index = sessions.indexOf(session);
+        if (index > -1) {
+            sessions.splice(index, 1);
+        }
+        this.sessions = sessions;
+        let url = this.connection.baseUrl + "leaveSession.json?connection=" + this.connection.connectionToken + "&localPlayer=" + this.localPlayerToken + "&session=" + session.getSessionToken();
+        return fetch(url)
+        .then(response => response.json());
+    }
+};
+
+class MPConnection {
+    constructor(gameUUID, deviceUUID, baseUrl) {
+        this.gameUUID = gameUUID;
+        this.deviceUUID = deviceUUID;
+        this.baseUrl = baseUrl;
+
+        this.connectionToken = null;
+        this.localPlayer = null;
+    }
+
+    /**
+     * Connect to the game's instance of the multiplay service.
+     * @return {Promise} a promise that is fulfilled when the game is connected.
+     */
+    connect() {
+        let connectUrl = this.baseUrl + "connect.json?game=" + this.gameUUID;
+        return fetch(connectUrl)
+        .then(response => response.json())
+        .then(data => {
+            this.connectionToken = data.connectionToken;
+        });
+    }
+
+    /**
+     * Login the local player.
+     * If the player has recently been logged in on this device this may return a
+     * logged in an authorized player.
+     * If the current device has no memory of the last time this player logged in,
+     * or the server decides it can't trust the device an unathorized guest player
+     * is returned.
+     * You need to authorize the player to convert them from a guest into a previously
+     * created authorized user.
+     * @return {Promise} a promise that is fulfilled when player is logged in.
+     */
+    login() {
+        let loginUrl = connection.baseUrl + "login.json?connection=" + this.connectionToken + "&localDevice=" + this.deviceUUID;
+        return fetch(loginUrl)
+        .then(response => response.json())
+        .then(data => {
+            let localPlayerToken = data.localPlayerToken;
+            let displayName = data.displayName;
+            let friendCode = data.friendCode;
+            let authenticated = data.authenticated ? true : false;
+            let localPlayer = new MPLocalPlayer(this, localPlayerToken, displayName, friendCode, authenticated);
+            this.localPlayer = localPlayer;
+            return localPlayer;
+        });
+    }
+
+    getLocalPlayer() {
+        return this.localPlayer;
+    }
+};
+
+/**
+ * Concrete subclass of MPConnection to use when connection from a browser client with the 'window' context
+ */
+class MPBrowserConnection extends MPConnection {
```

Comment:
Yet again, this is just for JavaScript. Look at the java Connection object instead.

Reply:
this file is left over from the initial js port, it's removed in the real file.

Reply:
removed in the actual file, this was checked-in in error